backvendor/vendored.go

```diff
@@ -131,20 +131,13 @@ func updateHashesAfterStrip(hashes *FileHashes, wt *WorkingTree, ref string, pat
 		defer os.Remove(f.Name())
 
 		// Write to the file and close it, checking for errors
-		err = func() (err error) {
-			defer func() {
-				cerr := f.Close()
-				if err == nil {
-					// Write didn't fail but Close did
-					err = cerr
-				}
-			}()
-
-			_, err = w.WriteTo(f)
-			return err
-		}()
-
+		_, err = w.WriteTo(f)
 		if err != nil {
+			f.Close() // ignore any secondary error
+			return true, errors.Wrap(err, "updating hash")
```

Comment:
Why are you returning `true` directly? How about returning `anyChanged` instead?